src/smp/ipi.c

```diff
@@ -65,7 +69,10 @@ void ipiStallCoreCallback(bool_t irqPath)
          * handle the pending interrupt. Its valid as interrups are async events! */
         SCHED_ENQUEUE_CURRENT_TCB;
         switchToIdleThread();
-
+#ifdef CONFIG_KERNEL_MCS
+        commitTime();
+        NODE_STATE(ksCurSC) = NODE_STATE(ksIdleThread)->tcbSchedContext;
```

Comment:
Is there a reason these can't be replaced with switchSchedContext() which should do the same thing after a call to switchToIdleThread() but is more likely stay up to date with any edge cases?

Reply:
One reason is that we need to ensure that ksReprogram is left as false because we aren't programming a timeout (assuming that the core stall will complete in << KERNEL_WCET). switchSchedContext sets ksReprogram to true and so we would need to set it to false right after.